receiver/fluentforwardreceiver/collector.go

```diff
@@ -53,35 +53,21 @@ func (c *Collector) processEvents(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case e := <-c.eventCh:
-			buffered := []Event{e}
+			out := plog.NewLogs()
+			rls := out.ResourceLogs().AppendEmpty()
+			logSlice := rls.ScopeLogs().AppendEmpty().LogRecords()
+			e.LogRecords().MoveAndAppendTo(logSlice)
+
 			// Pull out anything waiting on the eventCh to get better
 			// efficiency on LogResource allocations.
-			buffered = fillBufferUntilChanEmpty(c.eventCh, buffered)
-
-			logs := collectLogRecords(buffered)
-			_ = c.nextConsumer.ConsumeLogs(ctx, logs)
-		}
-	}
-}
+			select {
+			case e2 := <-c.eventCh:
+				e2.LogRecords().MoveAndAppendTo(logSlice)
+			default:
+			}
```

Comment:
Why `for {}` is dropped here? Doesn't it mean that we fetch only one event from the channel now instead of all?

Reply:
Because of the outer for {} loop we'll keep processing after every call to `c.nextConsumer.ConsumeLogs(ctx, out)` right?

Reply:
I believe we still want to pass all available logs to `ConsumeLogs` at once

Reply:
Good catch.

Reply:
Done.